lib/hooks/useHoliday.ts

```diff
@@ -0,0 +1,66 @@
+import { useEffect, useState } from "react"
+import { ApiGetResponse } from "../types/hooks"
+
+type Holiday = {
+    date: string
+    description: string
+    summary: string
+}
+
+interface IData extends Holiday {
+    start: {
+        date: string
+    }
+}
+
+function parseEvents(data: { items: IData[] }): Holiday[] {
+    const { items } = data
+    const holidays = []
+    for (const item of items) {
+        const { date } = item.start
+        const { description } = item
+        const { summary } = item
+        holidays.push({ date, description, summary })
+    }
+    return holidays
+}
+
+export const useHoliday = (): ApiGetResponse<Holiday[]> => {
+    const [response, setResponse] = useState<ApiGetResponse<Holiday[]>>({
+        isSuccess: false,
+        isLoading: true,
+        isError: false,
+    })
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                const apiResponse = await fetch(
+                    // eslint-disable-next-line max-len
+                    "https://www.googleapis.com/calendar/v3/calendars/fi.finnish%23holiday%40group.v.calendar.google.com/events?key=AIzaSyAMM0ZrGxYU4wYIfA2oO659iaitt96gPzI"
```

Comment:
Apikey should be considered a secret and not committed to git in plaintext. Please remove it, create a new apikey instead of this compromised one, and provide it as an environment variable.

Reply:
Also different environments should use different apikeys